### PR TITLE
Fix CI warnings and issues

### PR DESCRIPTION
**Pull request by xgouchet**

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -61,10 +59,6 @@ gradlePlugin {
     }
 }
 
-tasks.withType<KotlinCompile> {
```
Comment (0xnm) [inline buildSrc/build.gradle.kts:64]: we don't need it anymore?

File: samples/variants/build.gradle [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
-//apply plugin: 'com.datadoghq.dd-sdk-android-gradle-plugin'
```
Comment (0xnm) [inline samples/variants/build.gradle:3]: it seems this change should be reverted, this is a valid id per https://github.com/DataDog/dd-sdk-android-gradle-plugin/blob/0c19f89d19fbbe1de91cd0580b0f2b9fc3cdbf1b/dd-sdk-android-gradle-plugin/build.gradle.kts#L70

File: samples/variants/src/main/java/com/datadog/example/variants/MainActivity.kt [hunk 3]
```diff
@@ -1,9 +1,8 @@
-package com.example.variants
+package com.datadog.example.variants
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
-import com.example.lib.Placeholder // ktlint-disable no-unused-imports  unused import is on purpose
```
Comment (0xnm) [inline samples/variants/src/main/java/com/datadog/example/variants/MainActivity.kt:5]: need to revert this line

File: buildSrc/build.gradle.kts (hunk 1 above)
Comment (xgouchet) [reply buildSrc/build.gradle.kts:64]: No, it actually break the ci as the jvm running gradle uses JDK 17

**Review by 0xnm — APPROVED** [review]

**Pull request merged**